src/pages/SignUpPage/SignUpPage.tsx

```diff
@@ -0,0 +1,342 @@
+      <Box component="form" onSubmit={handleSubmit} sx={getFormStyles(theme)}>
+        <Box sx={getLogoStyles(theme)}>
+          <Logo orientation="horizontal" width="100%" height="100%" alt="Logo" />
+        </Box>
+
+        <FormInput
+          label="Email"
+          value={email}
+          onChange={(e) => {
+            setEmail(e.target.value);
+            setErrors({ ...errors, email: '' });
+          }}
+          placeholder="Enter your email"
+          errorMessage={errors.email}
+          sx={getInputStyles(122.42, theme)}
+        />
+
+        <FormInput
+          label="Username"
+          value={username}
+          onChange={(e) => {
+            setUsername(e.target.value);
+            setErrors({ ...errors, username: '' });
+          }}
+          placeholder="Enter your username"
+          errorMessage={errors.username}
+          sx={getInputStyles(225.65, theme)}
+        />
+
+        <FormInput
+          label="Password"
+          type="password"
+          value={password}
+          onChange={(e) => {
+            setPassword(e.target.value);
+            setErrors({ ...errors, password: '' });
+          }}
+          placeholder="Enter your password"
+          errorMessage={errors.password}
+          sx={getInputStyles(328.88, theme)}
+        />
+
+        <PrimaryButton
+          sx={getButtonStyles(theme)}
+          onClick={handleSubmit}
+          loading={loading}
+          disabled={isDisabled}
+        >
+          {buttonText}
+        </PrimaryButton>
+
+        <Typography sx={haveAccountTextStyles(theme)}>Have an Account?</Typography>
+
+        <Link href="/login" sx={loginLinkStyles(theme)}>
+          {loginText}
+        </Link>
+
+        <Typography sx={signupWithTextStyles(theme)}>{signupWithText}</Typography>
+
+        {showSocialIcons && (
+          <>
+            <Box
+              component="img"
+              src={GoogleIcon}
+              alt="Google"
+              sx={getSocialIconStyles(0, theme)}
+              onClick={() => window.open(googleLoginUrl, '_blank')}
+              onError={handleImgError}
+            />
+
+            <Box
+              component="img"
+              src={GithubIcon}
+              alt="GitHub"
+              sx={getSocialIconStyles(1, theme)}
+              onClick={() => window.open(githubLoginUrl, '_blank')}
+              onError={handleImgError}
+            />
+
+            <Box
+              component="img"
+              src={FacebookIcon}
+              alt="Facebook"
+              sx={getSocialIconStyles(2, theme)}
+              onClick={() => window.open(facebookLoginUrl, '_blank')}
+              onError={handleImgError}
+            />
+          </>
+        )}
+
+        {showLegalText && (
+          <Typography sx={legalTextStyles(theme)}>
+            This site is protected by reCAPTCHA and the Google{' '}
+            <Link href={privacyPolicyUrl} target="_blank" sx={legalLinkStyles}>
```

Comment:
External link opens in a new tab (target="_blank") but lacks rel="noopener noreferrer". This is a tabnabbing security risk. Always include rel="noopener noreferrer" when using target="_blank".

**Sugerencia:**
```typescript
Add rel="noopener noreferrer" to both external links, e.g. <Link href={privacyPolicyUrl} target="_blank" rel="noopener noreferrer" sx={legalLinkStyles}>
```